Update model reproducibility notebook

leilayesufu opened the pull request:
Good morning @GemmaTuron 
i have created the 03_model_reproducibility.ipynb file for model reproducibility. All the files used for the model_reproducibility has been added in the data folder inside a folder named model_reproducibility and the README.md file has been updated.

leilayesufu committed:
commit

leilayesufu committed:
commit

leilayesufu committed:
commit

leilayesufu committed:
commit

GemmaTuron commented:
Hi @leilayesufu this is great
Next steps:
- Report the findings on the README file of the /hERG folder
- eos30f3: test on CodeSpaces - collect the log file if it cannot be fetched to understand what might be happening
- Use Sarima's dataset for assessing model bias
- Add useful functions into an /src folder

leilayesufu committed:
commit

DhanshreeA commented:
Hi @leilayesufu happy to merge this PR after you:

- [x] merge/rebase it with main (since I've merged Sarima's PR today)
- [ ] and update the README with the findings. 

You can work on the other things in a new PR. :rocket:

leilayesufu committed:
Merge branch 'ersilia-os:main' into main

leilayesufu committed:
commit

leilayesufu committed:
commit

leilayesufu commented:
I have updated the PR with the following changes. 3 functions have been added to the src folder in the root directory.

performance_metrics_and_auroc_curve.py - This function get the performance metrics and plots the roc curve.  It takes in the predictions, the true labels and the model name.
process_inchikeys.py  - This function takes the smiles and activity column from the dataset, processes it and get the inchikey and returns. three columns containing smiles, inchikey and activity
scatter_plot_for_model_bias.py- This plots the scatter diagram of a given model's predictions.


I have also updated the notebooks to use the functions instead of repeated code. 
I have updated the README.md file with the results and tested for model bias with the 1000 dataset Sarima curated.

leilayesufu committed:
commit

DhanshreeA commented:
Hi @leilayesufu all the links you have provided within the README (results from the models' respective publications) are broken. Could you please revise those?

leilayesufu commented:
I have updated it.

DhanshreeA commented:
About the code refactoring: that's a good start! However, naming our files so specifically is making us miss out on the opportunity to extend these files in the future. Let's think about it this way, there are three top level tasks within any model validation that we have encountered so far:

- pre-processing data: checking if a smile is valid, if a smile is canonical, getting an inchi key for a given smile, etc.
- plotting results: plotting model results for a reference lib, or for the original data from a publication
- evaluating results: ie evaluating the model's results 

So I recommend naming the files as follows:
- `scatter_plot_for_model_bias.py` -> `plot.py`
- `process_inchikeys.py` -> `preprocess.py`
- `performance_metrics_and_auroc_curve.py` -> `evaluation.py`

Additionally, please work with the notebook that @Richiio has updated [here](https://github.com/leilayesufu/model-validations/blob/main/notebooks/00_general_data_processing.ipynb) such that the functions you write can be used both in your notebook, as well as in this notebook.

leilayesufu committed:
commit

leilayesufu commented:
@DhanshreeA I have updated the PR
